cmd/kube-controller-manager/app/autoscaling.go

```diff
@@ -85,7 +84,7 @@ func startHPAControllerWithMetricsClient(ctx ControllerContext, metricsClient me
 		ctx.ComponentConfig.HPAController.HorizontalPodAutoscalerTolerance,
 	)
 	go podautoscaler.NewHorizontalController(
-		hpaClientGoClient.CoreV1(),
+		hpaClient.CoreV1(),
```

Comment:
Will this make replicaCalc, scaleKindResolver and NewHorizontalController share the same client QPS quota?

Reply:
Yes but the ScaleKindResolver only make O(number of autoscaled resource kinds) calls so it should not be an issue.